spring-amqp/src/main/java/org/springframework/amqp/support/converter/DefaultClassMapper.java

```diff
@@ -186,4 +199,48 @@ public void fromClass(Class<?> clazz, MessageProperties properties) {
 		return toClass(classId);
 	}
 
+	private Class<?> toClass(String classId) {
+		if (this.idClassMapping.containsKey(classId)) {
+			return this.idClassMapping.get(classId);
+		}
+		if (classId.equals(DEFAULT_HASHTABLE_TYPE_ID)) {
+			return this.defaultMapClass;
+		}
+		try {
+			Class<?> aClass = ClassUtils.forName(classId, getClass().getClassLoader());
```

Comment:
I think this is good concern and should not be so hard to redo, but at the same time that is what we have in SI:
```
DeserializationConfig config = (DeserializationConfig) context.getConfig();
JavaType result = this.delegate.typeFromId(context, id);

Package aPackage = result.getRawClass().getPackage();
if (aPackage == null || isTrustedPackage(aPackage.getName())) {
		return result;
}
```

At least that is how we copied/pasted from @rwinch 😄 : https://github.com/spring-projects/spring-security/blob/master/core/src/main/java/org/springframework/security/jackson2/SecurityJackson2Modules.java#L180